src/planscape/forsys/generate_projects_for_a_single_scenario.R

```diff
@@ -5,84 +5,98 @@ suppressMessages({
-                  forsys::combine_priorities(
-                    fields = priority_impact_fields, 
-                    weights = priority_weights, 
-                    new_field = wp_str)
-  )
+      forsys::combine_priorities(
+        fields = priority_impact_fields,
+        weights = priority_weights,
+        new_field = wp_colname))
+
   # Parses wkt in the geo_wkt column and adds it to a "geometry" column.
-  # Patchmax expects column name to be "geometry" - do not change the variable
-  # name.
-  # Internally, Patchmax calls st_sf on geometry column values. This results in
-  # an error unless st_as_sfc is called on the wkt first.
-  geometry <- lapply(forsys_input_data[geo_wkt_field], st_as_sfc)
-  forsys_input_data <- cbind(forsys_input_data, geometry)
+  forsys_input_data_formatted <- forsys_input_data %>%
+    rename("geometry" = geo_wkt_field) %>%
+    st_as_sf(wkt = "geometry")
 
-  shp <- st_as_sf(forsys_input_data)
+  # k-means clustering
+  if (PreForsysClusterType == 'KMEANS') {
+    # k-means clustering execution
+    source("forsys/spatial_clustering.R")
+    forsys_input_data_formatted <- cluster_cells_to_stands(
+      stand_data = forsys_input_data_formatted,
+      scenario_priorities = c(wp_colname, priorities))
+  }
 
   # TODO: optimize project area generation parameters, SDW, EPW, sample_frac.
-  suppressMessages (
+  suppressMessages(
     run_outputs <- forsys::run(
       return_outputs = TRUE,
       write_outputs = enable_debug,
-      scenario_name = output_scenario_name,
-      scenario_write_tags = output_scenario_tag,
-      stand_data = shp,
+      stand_data = forsys_input_data_formatted,
+      stand_area_field = stand_area_field,
+      # TODO: do we want ownership or operability here?
+      stand_threshold = paste(stand_area_field, " > 0"),
       stand_id_field = stand_id_field,
       proj_id_field = proj_id_field,
-      stand_area_field = stand_area_field,
-      scenario_priorities = c(wp_str),
-      scenario_output_fields = c(priorities,
-                                 stand_area_field,
-                                 stand_cost_field),
+      scenario_name = output_scenario_name,
+      scenario_write_tags = output_scenario_tag,
+      scenario_priorities = wp_colname,
+      scenario_output_fields = c(
+        priorities,
+        stand_area_field,
+        # TODO: seems we're missing the cost field?
+        stand_cost_field),
       run_with_patchmax = TRUE,
       # target area per project? TODO: clarify what this does, and clarify
       # whether there's also a target cost per project.
-      patchmax_proj_size = 30000, 
-       # number of projects - TODO: clarify whether this should be a user input.
+      patchmax_proj_size = 20,
+      # number of projects - TODO: clarify whether this should be a user input.
       patchmax_proj_number = 3,
-      patchmax_SDW = 0.5,
-      patchmax_EPW = 0.5, 
-      patchmax_sample_frac = 0.01,
+      patchmax_SDW = 1,
```

Comment:
Once we have realistic planning areas and configs from users, my goal is to run patchmax with several different parameters, present results to users, and see which project areas they prefer.

So it's fine to change the defaults, but just be aware that that's the goal.

Reply:
Acknowledged.